src-front/src/components/WordCloud/WordCloud.tsx

```diff
@@ -0,0 +1,241 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { Application } from "pixi.js";
+import * as PIXI from "pixi.js";
+import {
+  Vector,
+  Point,
+  calculateVec,
+  calculateVecSquared,
+} from "../../lib/vectorsUtility";
+
+interface Word {
+  text: string;
+  position: Vector;
+  speed: Vector;
+  pixiText: PIXI.Text;
+  mass: number;
+}
+
+export default function WordCloud() {
+  const refCanvas = useRef<HTMLDivElement>(null);
+  const refContainer = useRef<HTMLDivElement>(null);
+  const [containerSize, setContainerSize] = useState({ width: 0, height: 0 });
+  const refWords = useRef<Word[]>([]);
+  const [inputValue, setInputValue] = useState("");
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setInputValue(event.target.value);
+  }
+
+  function deleteWord(index: number) {
+    refWords.current[index].pixiText.removeFromParent();
+    refWords.current.splice(index, 1);
+  }
+
+  const updateSpeed = useCallback(
+    (word: Word) => {
+      const center: Point = {
+        x: containerSize.width / 2,
+        y: containerSize.width / 2,
+      };
+
+      const centerForce = calculateVec(
+        word.position.x,
+        word.position.y,
+        center.x,
+        center.y,
+        0.5 * word.mass,
+      );
+      word.speed = centerForce;
+
+      refWords.current.forEach((currWord) => {
```

Comment:
If performance remains a big issue, trying different type of loops (`for` vs `forEach`) could be an option, although it is very unlikely that there is much difference. (To explore as a last resort.)